src/test/java/nl/tudelft/ewi/devhub/webtests/views/AssignmentView.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    public Assignment getAssignment() throws ParseException {
+        return new Assignment(getDriver().findElement(By.cssSelector("a[href^=\"deliveries\"]")));
+    }
+
+    @Data
+    public class Assignment {
+        @Getter(AccessLevel.NONE)
+        private final WebElement anchor;
+        private final Review review = new Review();
+
+        public String getAuthor() {
+            return getDriver().findElements(By.tagName("dd")).get(0).getText();
+        }
+
+        public String getName() {
+            return getDriver().findElement(By.cssSelector("div .col-md-offset-2.col-md-10 h4")).getText();
+        }
+
+        public Delivery.State getStatus() {
+            return Delivery.State.valueOf(getDriver().findElement(By.cssSelector("span.label")).getText().toUpperCase());
+        }
+
+        public Date getDueDate() throws ParseException {
+            return new SimpleDateFormat("EEEE dd MMMM yyyy k:m", Locale.US).parse(
+                    getDriver().findElements(By.tagName("dd")).get(1).getText());
+        }
+
+        public DeliveryReviewView click() {
+            anchor.click();
+            return new DeliveryReviewView(getDriver());
+        }
+    }
+
+    @Data
+    public class Review {
+        public String getReviewer() {
+            final String footer = getFooter();
+            if (footer == null) {
+                return null;
+            }
+
+            return footer.split(" on ")[0].trim();
+        }
+
+        public double getGrade() {
+            final List<WebElement> ddElements = getDriver().findElements(By.cssSelector("blockquote dl dd"));
+            if (ddElements.size() == 2) {
+                ddElements.get(0).getText();
+            } else if (getDriver().findElement(By.cssSelector("blockquote dl dt")).getText().equalsIgnoreCase("Grade")) {
+                return Double.parseDouble(ddElements.get(0).getText());
+            }
+
+            return 0D;
+        }
+
+        @Nullable
+        public String getCommentary() {
+            final List<WebElement> ddElements = getDriver().findElements(By.cssSelector("blockquote dl dd"));
+            if (ddElements.size() == 2) {
+                ddElements.get(1).getText();
+            } else if (getDriver().findElement(By.cssSelector("blockquote dl dt")).getText().equalsIgnoreCase("Remarks")) {
+                    return ddElements.get(0).getText();
+            }
+
+            return null;
+        }
+
+
+        @Nullable
+        public Date getReviewTime() throws ParseException {
+            final String footer = getFooter();
+            if (footer == null) {
+                return null;
+            }
+
+            return new SimpleDateFormat("EEEE dd MMMM yyyy k:m", Locale.US).parse(footer.split(" on ")[1].trim());
+        }
+
+        @Nullable
+        private String getFooter() {
+            try {
+                return getDriver().findElement(By.cssSelector("footer.small")).getText();
```

Comment:
Catching exceptions in tests is actually quite bad! Two things can happen:
* The element was expected to be not existing, a false warning is being logged.
* The element was not present while it was expected, but the test still passes
Which behaviour is intended is very hard to read and more importantly, error prone.

Try:
```java
getDriver().findElements(By.id("btn-close")).stream().findAny().isPresent();
```